test/cellarea.jl

```diff
@@ -70,4 +71,29 @@ include(joinpath(dirname(pathof(Rasters)), "../test/test_utils.jl"))
     # test missing crs throws an error
     nocrsdimz = setcrs(dimz, nothing)
     @test_throws ArgumentError cellarea(nocrsdimz)
+
+    @testset "Unitful cellarea" begin
+         # Case 1: cellarea with unitful manifold
+         # This is the simplest case
+         unitful_manifold = Spherical(; radius = Spherical().radius * Unitful.u"m")
+         unitful_cellarea = cellarea(unitful_manifold, dimz_3857)
+         @test unitful_cellarea isa Raster{<:Unitful.Quantity}
+         @test Unitful.ustrip.(parent(unitful_cellarea)) == cellarea(Spherical(; radius = unitful_manifold.radius |> Unitful.ustrip), dimz_3857)
+         # Case 2: unitful dimensions
+         ux_3857 = rebuild(x_3857; val = rebuild(val(x_3857); data = val(x_3857) .* Unitful.u"m", span = Regular(val(x_3857).span.step * Unitful.u"m")))
+         uy_3857 = rebuild(y_3857; val = rebuild(val(y_3857); data = val(y_3857) .* Unitful.u"m", span = Regular(val(y_3857).span.step * Unitful.u"m")))
+         unitful_dimz_3857 = (ux_3857, uy_3857)
+         @test cellarea(Planar(), unitful_dimz_3857) isa Raster{<:Unitful.Quantity}
+         @test Unitful.ustrip.(parent(cellarea(Planar(), unitful_dimz_3857))) == parent(cellarea(Planar(), dimz_3857))
+         # Unitful lookups shouldn't work without a unitful manifold
+         @test_throws Unitful.DimensionError cellarea(unitful_dimz_3857)
+         # The tests below fail because Unitful apparently can't convert to Float64...
+         # (see https://github.com/PainterQubits/Unitful.jl/issues/742)
+         # But we also have to re-convert back to the original unit type, which cellarea currently
+         # doesn't do.
+
```

Comment:
```suggestion
         # This needs fixes for unit handling

```

Unitful is like a free test suite for physical models not converting is by design!

If we are going to make units work here we need to do `ustrip(u"m", x)` to ensure we are actually getting metres coming in. If people pass in kilometers and we convert back and forth with Proj.jl treating them as meters in the middle, we are really breaking the contract.

Reply:
That makes sense.  Maybe we have a `maybe_ustrip(unit, x)` defined in Rasters, its definition for Unitful `unit` in `RastersUnitfulExt`, and its definition for Proj units in `RastersProjUnitfulExt`?

(This is getting to be a mouthful..)

Reply:
Yeah those names! But makes sense. Somehow we need some mapping from Proj units to Unitful.

Reply:
Why does conversion from Proj to Unitful have to live in Rasters.jl? Shouldn't this be in a ProjUnitfulExt?

Reply:
Hmm, we could introduce a nice interface to batch transform an array, but do it out of place in Proj.jl

Reply:
I was initially going to put that here because this calls the C-API directly, but if it doesn't have to then there's no reason to stop this from calling Proj.

Reply:
I think we need a more general package for parsing geospatial units to Unitful. There is Proj, gdal, CF conventions, and some overlap between them.

Maybe UnitfulGeo.jl and it can go in JuliaGeo